### ISSUE TITLE
Cannot screen share in Linux Mint 21 (Ubuntu 22.04)

### ISSUE THREAD

**Issue by gitgt** [mattermost/mattermost-plugin-calls#464]
This may be explained somewhere, but I did not find in Troubleshooting or search here. 
After some problems, now I have plugin working in Linux desktop client v5.4.0, call is fine. 
But I cannot screen share (never did), I get error in plugin that thre is no permission to share. 
Should a permisison be asked automatically or should I allow it manually?

**Comment by streamer45** [comment]
@gitgt If it's failing due to permissions there's not much we can do from the app. It's usually an operating system setting preventing you from sharing. Could you try connecting from a browser and see if you can share?

**Comment by gitgt** [comment]
I can share from the browser, after I allow. I would guess MM can do the same, ask for permission.

**Comment by streamer45** [comment]
> I can share from the browser, after I allow and select window. I would guess MM should do the same, ask for permission.

@gitgt I think this may be an operating system compatibility issue. I assume you are running on Wayland.

Something similar to https://askubuntu.com/questions/1405195/how-to-share-a-screen-in-ms-teams-or-zoom-from-ubuntu-22-04 and https://github.com/IsmaelMartinez/teams-for-linux/issues/629

I'll see if there's anything we can do from the app side but I am not very confident.

**Comment by gitgt** [comment]
@streamer45 Linux Mint is X11, not Wayland.

**Comment by streamer45** [comment]
Alright. Can you try again to reproduce and share the app logs (preferably at debug level)?

**Comment by gitgt** [comment]
Can you please explain how to run debug and where the logs would be, are Mattermost and the plugin logs together?

**Comment by streamer45** [comment]
> Can you please explain how to run debug and where the logs would be, are Mattermost and the plugin logs together? Is it some in-buillt option or I need to compile, as now I installed from .deb?

At the moment I am only interested in the desktop app logs, see https://docs.mattermost.com/install/troubleshooting.html#mattermost-desktop-app-logs for details.

Debug level can be enabled from the Settings panel (three dots -> File -> Settings):

![image](https://github.com/mattermost/mattermost-plugin-calls/assets/1832946/7928d499-066f-44ce-aa31-22581cf834d8)

**Comment by gitgt** [comment]
Here it is, thanks.
[main.log](https://github.com/mattermost/mattermost-plugin-calls/files/12024732/main.log)

**Comment by gitgt** [comment]
Yes, I select window to share and then it fails. Sorry for not being precise.
![image](https://github.com/mattermost/mattermost-plugin-calls/assets/7240617/19eda41d-96ac-498c-8f01-f26a59675f0f)

When I run MM_DEBUG_CALLS_WIDGET=true mattermost-desktop %U I do not get a window, so I have to click and icon. 
But I see browser debug window. I hope this is it.

[Share.log](https://github.com/mattermost/mattermost-plugin-calls/files/12029188/Share.log)

**Comment by streamer45** [comment]
@gitgt Thanks that's it. Looks like it's failing to access an audio source. Do you have the experimental flags turned on? That may explain this as it will try to share system audio but it won't work on Linux. Could you try issuing the `/call experimental off` command in a channel and attempt again to share?

**Comment by streamer45** [comment]
> That is it, works now. I did not know about Experimental. Can I turn it off permanently? Thanks

Unless you explicitly turn them on again with the `/call experimental on` they will be kept off indefinitely so you should be good now.

**Comment by gitgt** [comment]
Oh, this seems a dupe of https://github.com/mattermost/mattermost-plugin-calls/issues/131.
And is related to https://github.com/mattermost/mattermost-plugin-calls/issues/127.
